src/qmozview_p.cpp

```diff
@@ -433,54 +418,31 @@ void QGraphicsMozViewPrivate::sendAsyncMessage(const QString &name, const QVaria
     mView->SendAsyncMessage((const char16_t*)name.constData(), NS_ConvertUTF8toUTF16(array.constData()).get());
 }
 
-void QGraphicsMozViewPrivate::UpdateViewSize()
+void QMozViewPrivate::setMozWindow(QMozWindow* window)
 {
-    if (mSize.isEmpty())
-        return;
-
-    if (!mViewInitialized) {
-        return;
-    }
-
-    if (mContext->GetApp()->IsAccelerated() && mHasContext) {
-        mView->SetGLViewPortSize(mGLSurfaceSize.width(), mGLSurfaceSize.height());
-    }
-    mView->SetViewSize(mSize.width(), mSize.height());
-
-    if (mOrientationDirty) {
-        ScreenRotation rotation = ROTATION_0;
-        switch (mOrientation) {
-        case Qt::LandscapeOrientation:
-            rotation = mozilla::ROTATION_90;
-            break;
-        case Qt::InvertedLandscapeOrientation:
-            rotation = mozilla::ROTATION_270;
-            break;
-        case Qt::InvertedPortraitOrientation:
-            rotation = mozilla::ROTATION_180;
-            break;
-        default:
-            break;
+    mMozWindow = window;
+    if (mMozWindow) {
+        if (!mSize.isEmpty()) {
+            mMozWindow->setSize(mSize.toSize());
         }
-        mView->SetScreenRotation(rotation);
-        mOrientationDirty = false;
+        connect(mMozWindow.data(), &QMozWindow::compositorCreated,
+                this, &QMozViewPrivate::onCompositorCreated);
```

Comment:
Could done like below. The ```onCompositorCreated``` slot is just a setter and not needed in the private API. This way we'd have a few lines less in the private class.

```
connect(mMozWindow.data(), &QMozWindow::compositorCreated, this, [=]() {
    mHasCompositor = true;
});
```